File: uptmdigital_app/lib/models/anuncio.dart
class Anuncio {
  final int idAnuncio;
  final String titulo;
  final String contenido;
  final String fechaPublicacion;
  final String? autor;
  final int? usuarioId;
  final String prioridad;

  Anuncio({
    required this.idAnuncio,
    required this.titulo,
    required this.contenido,
    required this.fechaPublicacion,
    this.autor,
    this.usuarioId,
    this.prioridad = "Normal",
  });

  factory Anuncio.fromJson(Map<String, dynamic> json) {
    return Anuncio(
      idAnuncio: json['idAnuncio'] ?? 0,
      titulo: json['titulo'] ?? '',
      contenido: json['contenido'] ?? '',
      fechaPublicacion: json['fechaPublicacion'] ?? '',
      autor: json['autor'],
      usuarioId: json['usuarioId'],
      prioridad: json['prioridad'] ?? 'Normal',
    );
  }
}

File: UPTMDigital.API/Models/Anuncio.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace UPTMDigital.API.Models
{
    [Table("Anuncio")]
    public class Anuncio
    {
        [Key]
        public int IdAnuncio { get; set; }
        public string Titulo { get; set; } = null!;
        public string Contenido { get; set; } = null!;
        public DateTime FechaPublicacion { get; set; }

        // --- FK Normalizada ---
        public int? UsuarioId { get; set; }
        [ForeignKey("UsuarioId")]
        public Usuario? Usuario { get; set; }

        /// <summary>Nombre del autor (cache para mostrar sin JOIN).</summary>
        public string? Autor { get; set; }

        // --- Nuevos Campos Fase 6 (Cartelera Jerárquica) ---

        // Si es null, va dirigido a todos
        public int? CarreraId { get; set; }
        [ForeignKey("CarreraId")]
        public Carrera? Carrera { get; set; }

        // Si es null, va dirigido a todos los roles
        public int? RolId { get; set; }
        [ForeignKey("RolId")]
        public Rol? Rol { get; set; }

        // Prioridad: Normal, Urgente, Critica
        public string Prioridad { get; set; } = "Normal";

        // Trimestre dirigido (Opcional)
        public string? Trimestre { get; set; }

        public bool Activo { get; set; } = true;
    }
}
